### PR TITLE
[Term Entry] Python: defaultdict

### LINKED ISSUE #4232: [Entry] Python: defaultdict()
### Reason/inspiration (optional)

We would like a new term entry on the `defaultdict()` method for [Collection Module](https://www.codecademy.com/resources/docs/python/collections-module). The entry should go in a new file under `docs/content/python/concepts/collections-module/terms/defaultdict/defaultdict.md`.

The entry should have the following:
– An introductory paragraph about what the method does.
– A syntax section with a pseudo code block explaining it.
– An example showing the method in use.
– A codebyte of the method.

Please refer to the [term entry template](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [x] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by Sriparno08**
### Description

- Added new term entry for the 'defaultdict' datatype in Python
- Closes #4232 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/python/concepts/collections-module/terms/defaultdict/defaultdict.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
+---
+Title: 'collections.defaultdict'
```
Comment (dakshdeepHERE) [inline content/python/concepts/collections-module/terms/defaultdict/defaultdict.md]:
```suggestion
Title: 'Defaultdict'
```

File: content/python/concepts/collections-module/terms/defaultdict/defaultdict.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
+---
+Title: 'Defaultdict'
```
Comment (ishg-153) [inline content/python/concepts/collections-module/terms/defaultdict/defaultdict.md:2]:
```suggestion
Title: 'defaultdict'
```

File: content/python/concepts/collections-module/terms/defaultdict/defaultdict.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+Tags:
+  - 'Modules'
+  - 'Dictionaries'
+  - 'Data Types'
+CatalogContent:
+  - 'learn-python-3'
+  - 'paths/computer-science'
+---
+
+In Python, **defaultdict** is a [data type](https://www.codecademy.com/resources/docs/python/data-types) that belongs to the [`collections`](https://www.codecademy.com/resources/docs/python/collections-module) [module](https://www.codecademy.com/resources/docs/python/modules). It is a [`dict`](https://www.codecademy.com/resources/docs/python/dictionaries) subclass that is used to return a dictionary-like object.
```
Comment (ishg-153) [inline content/python/concepts/collections-module/terms/defaultdict/defaultdict.md:16]:
```suggestion
In Python, **defaultdict** is a data type that belongs to the [`collections`](https://www.codecademy.com/resources/docs/python/collections-module) module. It is a [`dictionary`](https://www.codecademy.com/resources/docs/python/dictionaries) subclass that is used to return a dictionary-like object.
```

File: content/python/concepts/collections-module/terms/defaultdict/defaultdict.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+## Syntax
+
+```pseudo
+defaultdict(default_factory)
```
Comment (ishg-153) [inline content/python/concepts/collections-module/terms/defaultdict/defaultdict.md:21]:
```suggestion
collections.defaultdict(default_factory)
```

File: content/python/concepts/collections-module/terms/defaultdict/defaultdict.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+```
+
+- `default_factory`: It is a [function](https://www.codecademy.com/resources/docs/python/functions) that returns the default value for the dictionary specified.
```
Comment (ishg-153) [inline content/python/concepts/collections-module/terms/defaultdict/defaultdict.md:24]:
```suggestion
- `default_factory`: It gives the default value for the dictionary object.
```

File: content/python/concepts/collections-module/terms/defaultdict/defaultdict.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+```shell
+100
+90
+Not Declared
+```
+
+## Codebyte Example
+
+Here is a codebyte example for the `defaultdict` data type:
```
Comment (ishg-153) [inline content/python/concepts/collections-module/terms/defaultdict/defaultdict.md:56]:
```suggestion
Run the following codeblock and explore more about the `defaultdict` data type:
```

**Review by ishg-153 — CHANGES_REQUESTED** [review]
@Sriparno08 I have suggested a couple of changes. Let me know when you are done making them.

**Comment by Sriparno08** [comment]
Done, @ishg-153 👍

**Review by ishg-153 — APPROVED** [review]
@Sriparno08 Great! This is ready to merge.

**Pull request merged**